### PR TITLE
PMM-8118 Go modules

### PR DESCRIPTION
**Pull request by Dasio**
https://jira.percona.com/browse/PMM-8118

FB: https://github.com/Percona-Lab/pmm-submodules/pull/1788

Vendor folder had also .proto files for 3rd party. It is not possible to do it without vendor files. (Also prototool seems to be deprecated in favor of  https://buf.build/). So in `make init` is manual download of required proto files.

We already got v2 tags.  Based on golang semantic versioning, we should add /v2 prefix to import path?

### CODE REVIEW

**Review by BupycHuk — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /.idea/
 /.vscode/
 
-/vendor/
 /api/prototool
 /api/nginx/*.pem
+/bin
+/third_party
```
Comment (oter) [inline .gitignore:7]: let's a new line

File: tools/tools.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	_ "github.com/mwitkow/go-proto-validators/protoc-gen-govalidators"
+	_ "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway"
+	_ "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger"
+	_ "github.com/go-openapi/runtime/client"
+	_ "github.com/go-openapi/spec"
+	_ "github.com/go-swagger/go-swagger/cmd/swagger"
+	_ "github.com/BurntSushi/go-sumtype"
+	_ "github.com/Percona-Lab/swagger-order"
+
+)
```
Comment (oter) [inline tools/tools.go:18]: new line

File: api/Makefile [hunk 3]
```diff
@@ -1,23 +1,37 @@
...
+
 help:                 ## Display this help message.
 	@echo "Please use \`make <target>\` where <target> is one of:"
 	@grep '^[a-zA-Z]' $(MAKEFILE_LIST) | \
 		awk -F ':.*?## ' 'NF==2 {printf "  %-26s%s\n", $$1, $$2}'
 
-init:                 ## Fill ../vendor/ and install tools.
+init:                 ## Install tools.
 	# https://github.com/uber/prototool#installation
 	curl -L https://github.com/uber/prototool/releases/download/v1.10.0/prototool-$(shell uname -s)-$(shell uname -m) -o ./prototool-new
```
Comment (oter) [inline api/Makefile:13]: can we install prototool using `go build` like below?

Comment (Dasio) [reply api/Makefile:13]: Getting error
```
	github.com/golang/protobuf/jsonpb/jsonpb_test_proto: module github.com/golang/protobuf@latest found (v1.5.2), but does not contain package github.com/golang/protobuf/jsonpb/jsonpb_test_proto
```
https://github.com/uber/prototool/issues/559

**Review by BupycHuk — APPROVED** [review]

**Review by oter — APPROVED** [review]

**Review by askomorokhov — APPROVED** [review]

**Comment by denisok** [comment]
@BupycHuk should we land this ? @Dasio please rebase

**Pull request merged**